### PR TITLE
Updated definitions for sax-js 

### PR DESCRIPTION
**Pull request by mtrivolli**
Updates to the sax type definitions to include the isSelfClosing attribute and to fix the fact that, when the xmlns option is enabled, the attributes property in the tag element is of type [key: string]: Attribute, being Attribute a similar interface to Tag.

As a consequence, the onopentag method is changed to onopentag(tag: Tag | QualifiedTag): void, where:

```
    interface Tag {
      name: string;
      isSelfClosing: boolean;
      attributes: { [key: string]: string };
    }

    interface QualifiedTag {
      name: string;
      prefix: string;
      local: string;
      uri: string;
      ns: { [key: string]: string };
      attributes: { [key: string]: Attribute };
    }

   interface Attribute {
      name: string;
      prefix: string;
      local: string;
      uri: string;
      value: string;
   }
```

### CODE REVIEW

File: sax/sax.d.ts [hunk 1]
```diff
@@ -16,15 +16,30 @@ declare module "sax" {
...
+        attributes: { [key: string]: Attribute };
+    }
 
-        // Available if opt.xmlns
-        ns?: { [key: string]: string };
-        prefix?: string;
-        local?: string;
-        uri?: string;
+    export interface Tag extends BaseTag {
+      attributes: { [key: string]: string };
```
Comment (vsiao) [inline sax/sax.d.ts]: please use consistent indentation (4 spaces)

Comment (mtrivolli) [inline sax/sax.d.ts]: Fixed

File: sax/sax-tests.ts [hunk 2]
```diff
@@ -17,7 +17,9 @@ parser.onerror = function(e: Error) {
 parser.ontext = function(text: string) {
 };
 
-parser.onopentag = function(tag: sax.Tag) {
```
Comment (vsiao) [inline sax/sax-tests.ts]: I'd recommend adding a test that asserts that properties in `tag.attributes` have type `string` for `sax.Tag`, and then something that reads the `QualifiedName` properties for `sax.QualifiedTag`. For example:

```ts
parser.onopentag = function(tag: sax.Tag) {
  var attribValue: string = tag.attributes["foo"];
};

// if opt.xmlns === true
parser.onopentag = function(tag: sax.QualifiedTag) {
  var prefix: string = tag.prefix;
  var attribPrefix: string = tag.attributes["foo"].prefix;
  var attribValue: string = tag.attributes["foo"].value;
};
```

File: sax/sax.d.ts [hunk 3]
```diff
@@ -16,15 +16,30 @@ declare module "sax" {
...
-    export interface Tag {
+    export interface QualifiedName {
         name: string;
-        attributes: { [key: string]: string };
+        prefix: string;
+        local: string;
+        uri: string;
+    }
+
+    export interface Attribute extends QualifiedName {
```
Comment (vsiao) [inline sax/sax.d.ts]: `QualifiedAttribute`?

File: sax/sax-tests.ts (hunk 2 above)
Comment (mtrivolli) [inline sax/sax-tests.ts]: You're right. Added.

File: sax/sax.d.ts (hunk 3 above)
Comment (mtrivolli) [inline sax/sax.d.ts]: Yes, it makes sense for the sake of consistency. Fixed.

**Comment by vsiao** [comment]
thanks, looks great :+1:

**Pull request merged**